### PR TITLE
**BREAKING** Change apiKey.getAll() to return all keys w/o any filter & add apiKey.getAllNamedUserApiKeys()

### PR DESCRIPTION
**Pull request by thgreasi**
The apiKeys.getAllNamedUserApiKeys will be offering
the same behavior.

Change-type: major
See: https://jel.ly.fish/287934d0-6fb2-4af2-888a-47966b9393f0
See: https://www.flowdock.com/app/rulemotion/resin-frontend/threads/aLBhe-Vfl7EBsb-oNp71Mm_TTUv
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: lib/models/api-key.ts [hunk 1]
```diff
@@ -110,10 +110,47 @@ const getApiKeysModel = function (
...
+		 */
+		async getAllNamedUserApiKeys(
+			options: BalenaSdk.PineOptions<BalenaSdk.ApiKey> = {},
+		): Promise<BalenaSdk.ApiKey[]> {
+			return await pine.get({
+				resource: 'api_key',
+				options: mergePineOptions(
+					{
 						$filter: {
+							is_of__actor: await sdkInstance.auth.getUserActorId(),
```
Comment (nitishagar) [inline lib/models/api-key.ts:150]: Do we want to do a check for valid API keys also? (Assuming we have a time check or something similar)

File: tests/integration/models/api-key.spec.ts [hunk 2]
```diff
@@ -196,13 +245,17 @@ describe('API Key model', function () {
...
-				return expect(balena.models.apiKey.revoke(this.apiKey.id))
-					.to.not.be.rejected.then(() => balena.models.apiKey.getAll())
-					.then(function (apiKeys) {
-						expect(apiKeys).to.be.an('array');
-						expect(apiKeys).to.have.lengthOf(0);
-					});
+			it('should be able to revoke an exising api key', async function () {
+				await expect(balena.models.apiKey.revoke(this.apiKey.id)).to.not.be
+					.rejected;
+				const apiKeys = await balena.models.apiKey.getAll({
```
Comment (nitishagar) [inline tests/integration/models/api-key.spec.ts:251]: Move this to a common method? I see this repeated @ 48, 68 and here (251).

File: lib/models/api-key.ts (hunk 1 above)
Comment (thgreasi) [reply lib/models/api-key.ts:150]: We atm don't have such time constrained validity periods, but even if we had I wouldn't embed such a check here.
I mean, this major change was required just b/c we had an embedded check, so the less the better.

File: tests/integration/models/api-key.spec.ts (hunk 2 above)
Comment (thgreasi) [reply tests/integration/models/api-key.spec.ts:251]: Indeed... It's called `getAllNamedUserApiKeys()`...

**Review by JSReds — APPROVED** [review]
nice one!

**Pull request merged**